flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -914,46 +918,46 @@ static constexpr RuntimeFunction pgmathPrecise[] = {
 };
 
 static mlir::FunctionType genF32F32FuncType(mlir::MLIRContext *context) {
-  auto t = mlir::FloatType::getF32(context);
+  mlir::FloatType t = mlir::FloatType::getF32(context);
```

Comment:
Is this really considered a problem? What else would calling a constructor on a `mlir::FloatType` return if not an `mlir::FloatType`? <sigh>

Reply:
Reverted those